kanban-all/kanban-data/src/main/java/ru/ifmo/p3411/repositories/TaskRepository.java

```diff
@@ -0,0 +1,14 @@
+package ru.ifmo.p3411.repositories;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.stereotype.Repository;
+import ru.ifmo.p3411.data.Task;
+
+/**
+ * @author Ellepsis
+ * @since 1.0
+ */
+@Repository
+public interface TaskRepository extends JpaRepository<Task, Long>, JpaSpecificationExecutor<Task> {
```

Comment:
I have added a basic smoke test with in-memory H2 datasource. Problems:

THE MOST F*** CRAZY:

ru.ifmo.p3411.data.SystemUser#kanbanUser
javax.persistence.PrimaryKeyJoinColumn

will give you nice:
Caused by: java.lang.NullPointerException
at org.hibernate.mapping.PersistentClass.createPrimaryKey(PersistentClass.java:363)

I don't even sure it's implemented by Hibernate.


ru.ifmo.p3411.data.TaskUser#task
ru.ifmo.p3411.data.TaskUser#kanbanUser

Gives

Caused by: org.hibernate.MappingException: Composite-id class must implement Serializable: ru.ifmo.p3411.data.BoardUserPermission
at org.hibernate.mapping.RootClass.checkCompositeIdentifier(RootClass.java:291)

and elsewhere composite id is used.